src/services/assetService.ts

```diff
@@ -171,10 +190,74 @@ export class AssetService {
             const json = await this.storageProvider.readText(fileName);
             return JSON.parse(json) as IAssetMetadata;
         } catch (err) {
-            return {
-                asset: { ...asset },
-                regions: [],
-            };
+            if (asset.type === AssetType.TFRecord) {
+                return {
+                    asset: { ...asset },
+                    regions: await this.getRegionsFromTFRecord(asset),
+                };
+            } else {
+                return {
+                    asset: { ...asset },
+                    regions: [],
+                };
+            }
         }
     }
+
+    private async getRegionsFromTFRecord(asset: IAsset): Promise<IRegion[]> {
+        const objectArray = await this.getTFRecordObjectArrays(asset);
+        const regions: IRegion[] = [];
+        const tags: string[] = [];
+        let tagPos = 0;
+
+        // Add Regions from TFRecord in Regions
+        for (let index = 0; index < objectArray.textArray.length; index++) {
+            tagPos = tags.findIndex((tag) => tag === objectArray.textArray[index]);
+            if (tagPos < 0 ) {
+                tagPos = tags.length;
+                tags.push(objectArray.textArray[index]);
+            }
+
+            regions.push({
+                id: shortid.generate(),
+                type: RegionType.Rectangle,
+                tags: [{
+                    name: objectArray.textArray[index],
+                    color: TagColors[tagPos],
+                }],
+                boundingBox: {
+                    left: objectArray.xminArray[index] * objectArray.width,
+                    top: objectArray.yminArray[index] * objectArray.height,
+                    width: (objectArray.xmaxArray[index] - objectArray.xminArray[index]) * objectArray.width,
```

Comment:
Why multiply? isn't width equal to `max - min` ?

Reply:
Same reason as above.  The way we (v1) store these points in TFRecords format is in terms of relative points expressed in 0.0<=1.0 range respect to the image width and height